src/dpl/src/Place.cpp

```diff
@@ -1185,15 +1187,35 @@ void Opendp::setGridPaddedLoc(Node* cell, const GridX x, const GridY y)
 }
 void Opendp::placeCell(Node* cell, const GridX x, const GridY y)
 {
+  const DbuX original_x = cell->getLeft();
+  const DbuY original_y = cell->getBottom();
+  const bool was_placed = cell->isPlaced();
   grid_->paintPixel(cell, x, y);
   setGridPaddedLoc(cell, x, y);
   cell->setPlaced(true);
   cell->setOrient(grid_->gridPixel(x, y)->sites.at(
       cell->getDbInst()->getMaster()->getSite()));
+  if (journal_) {
+    MoveCellAction action;
+    action.setNode(cell);
+    action.setWasPlaced(was_placed);
+    action.setOrigLocation(original_x, original_y);
+    action.setNewLocation(cell->getLeft(), cell->getBottom());
```

Comment:
Can this be done in the ctor and the class members be made const?  I believe this is immutable data.

Likewise the other actions.